_analyzers/custom-analyzer.md

```diff
@@ -0,0 +1,312 @@
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "phone_number_analyzer": {
+          "type": "custom",
+          "char_filter": ["phone_normalization"],
+          "tokenizer": "standard",
+          "filter": ["lowercase", "edge_ngram"]
+        }
+      },
+      "char_filter": {
+        "phone_normalization": {
+          "type": "pattern_replace",
+          "pattern": "[-\\s]",
+          "replacement": ""
+        }
+      },
+      "filter": {
+        "edge_ngram": {
+          "type": "edge_ngram",
+          "min_gram": 3,
+          "max_gram": 10
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Use the following request to examine the tokens generated using the analyzer:
+
+```json
+GET advanced_pattern_replace_analyzer_index/_analyze
+{
+  "analyzer": "phone_number_analyzer",
+  "text": "123-456 7890"
+}
+```
+{% include copy-curl.html %}
+
+The response contains the generated tokens:
+
+```json
+{
+  "tokens": [
+    {"token": "123","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "1234","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "12345","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "123456","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "1234567","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "12345678","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "123456789","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0},
+    {"token": "1234567890","start_offset": 0,"end_offset": 12,"type": "<NUM>","position": 0}
+  ]
+}
+```
+
+## Position increment gap
+
+The `position_increment_gap` parameter sets a positional gap between terms when indexing multi-valued fields, such as arrays. This gap ensures that phrase queries don't match terms across separate values unless explicitly allowed. For example, a default gap of 100 means terms in different array entries are 100 positions apart, preventing unintended matches in phrase searches. You can adjust this value or set it to `0` in order to allow phrases to span across array values.
+
+The following example demonstrates the effect of `position_increment_gap` using a `match_phrase` query.
+
+1. Index a document in a `test-index`:
+
+     ```json
+     PUT test-index/_doc/1
+     {
+       "names": [ "Slow green", "turtle swims"]
+     }
+     ```
+     {% include copy-curl.html %}
+
+1. Query the document using a `match_phrase` query:
+
+    ```json
+    GET test-index/_search
+    {
+      "query": {
+        "match_phrase": {
+          "names": {
+            "query": "green turtle" 
+          }
+        }
+      }
+    }
+    ```
+    {% include copy-curl.html %}
+    
+    The response returns no hits, because the distance between the terms `green` and `turtle` is `100` (the default `position_increment_gap`).
```

Comment:
```suggestion
    The response returns no hits because the distance between the terms `green` and `turtle` is `100` (the default `position_increment_gap`).
```